src/routes/services/passport-strategies.services.js

```diff
@@ -35,6 +49,25 @@ const facebookAuth = async (accessToken, refreshToken, profile, done) => {
   }
 };
 
+export const twitterAuth = async (token, tokenSecret, profile, done) => {
+  try {
+    const [user] = await User.findOrCreate({
+      where: { socialId: profile.id, socialProvider: 'twitter' },
+      defaults: {
+        firstName: profile.username,
+        username: profile.emails[0].value,
+        email: profile.emails[0].value,
+        socialProvider: profile.provider,
+      },
+    });
+
+    return done(null, user);
```

Comment:
I observe you are using both 'async await' and 'done' in this function. Are they doing different things?

Reply:
`Done` is used by passportJs during authentication.

Reply:
Okay. My thoughts are that `done` and `async await` do the same thing. Could we try using done alone without the async await? Or async await without the done? Let us see if any of the substitution will work.